syntax = "proto3";

package heartystore;

service HeartyStore {
  rpc Initialize(InitRequest) returns (InitResponse); 
  rpc Put(PutRequest) returns (PutResponse);
  rpc Get(GetRequest) returns (GetResponse);
  rpc List(ListRequest) returns (ListResponse);
  rpc Destroy(DestroyRequest) returns (DestroyResponse);
}

message InitRequest {
  string store_name = 1;
}

message InitResponse {
  bool success = 1;
  string message = 2;
}

message PutRequest {
  string store_name = 1;
  string file_identifier = 2;
  bytes file_content = 3; 
}

message PutResponse {
  bool success = 1;
  string message = 2;
}

message GetRequest {
  string store_name = 1;
  string file_identifier = 2;
}

message GetResponse {
  bool success = 1;
  string message = 2;
  bytes file_content = 3;
}

message ListRequest {}

message ListResponse {
  bool success = 1;
  string message = 2;
  repeated string store_names = 3;
}

message DestroyRequest {
  string store_name = 1;
}

message DestroyResponse {
  bool success = 1;
  string message = 2;
}